scripts/apps/authoring/authoring/directives/AuthoringDirective.js

```diff
@@ -138,6 +138,21 @@ export function AuthoringDirective(superdesk, superdeskFlags, authoringWorkspace
                 }
             });
 
+            // to fetch the updated item if there are associations
+            function getUpdatedItem() {
+                if ($scope.item && $scope.item.associations) {
+                    api.find('archive', $scope.item._id).then((item) => {
+                        $scope.updatedItem = item;
```

Comment:
Why are you getting the same item you already have opened and not related items by their IDs?

Reply:
The opened item contains only the ID of related items. So when publishing, instead of finding the related items one by one it's better to find the opened item with complete json of the related item once.

Reply:
> The opened item contains only the ID of related items.

[it doesn't seem so](https://user-images.githubusercontent.com/2076953/51377254-b9108000-1b0a-11e9-8cef-23d2fe24b8b3.gif). I see the entire payload in the associations.

Reply:
The data structure has to be the same all the time, not only when certain actions are performed.